### PR TITLE
Remove MaxParallelLevel

### PR DESCRIPTION
**Pull request by pdhaliwal-amd**

### CODE REVIEW

File: openmp/libomptarget/plugins/amdgpu/impl/data.cpp [hunk 1]
```diff
@@ -59,9 +59,9 @@ atmi_status_t Runtime::Malloc(void **ptr, size_t size, atmi_mem_place_t place) {
   hsa_amd_memory_pool_t pool = get_memory_pool_by_mem_place(place);
   hsa_status_t err = hsa_amd_memory_pool_allocate(pool, size, 0, ptr);
   ErrorCheck(atmi_malloc, err);
-  DEBUG_PRINT("Malloced [%s %d] %p\n",
-              place.dev_type == ATMI_DEVTYPE_CPU ? "CPU" : "GPU", place.dev_id,
-              *ptr);
+  // DEBUG_PRINT("Malloced [%s %d] %p\n",
+  //             place.dev_type == ATMI_DEVTYPE_CPU ? "CPU" : "GPU", place.dev_id,
+  //             *ptr);
```
Comment (pdhaliwal-amd) [inline openmp/libomptarget/plugins/amdgpu/impl/data.cpp:64]: Will remove these

**Comment by JonChesterfield** [comment]
The simplification is nice, but I thought malloc was broken on multiple gpu systems. It always allocates to device 0.

**Comment by pdhaliwal-amd** [comment]
I think you are true for malloc case. But shouldn't we be fixing the malloc then?

**Comment by ronlieb** [comment]
yes, if the malloc is broken, please fix it before landing this patch, would be my preference.,

**Comment by pdhaliwal-amd** [comment]
snap was failing due to missing free call after malloc. Its fixed now and snap is also working fine. However, this patch is slowing it down by a huge time. Total simulation time for snap is 60s with this patch against 20s without it.

**Comment by pdhaliwal-amd** [comment]
nekbone's performance is also worse. Having accumulated malloc before launching kernel should trivially be better than calling malloc each time before starting new nested loop. Unsure if this patch adds any value. I guess nvptx should also be slow for nested parallel for.

**Comment by JonChesterfield** [comment]
@saiislam <- this is great for the upstreaming effort, as it's simpler but slower

**Review by gregrodgers — APPROVED** [review]

**Comment by JonChesterfield** [comment]
I've pinged Pushpinder on this, will rebase it myself if I get around to it first

**Comment by JonChesterfield** [comment]
Squashed the two commits and rebased on aomp11, building locally now

**Comment by JonChesterfield** [comment]
Version currently on this patch passed testing locally. Snap slowed down as anticipated, other tests unchanged. Taking 25s (instead of 15) locally.

**Pull request merged**